src/middlewared/middlewared/api/v25_10_0/smb.py

```diff
@@ -206,8 +219,436 @@ class SmbSharePresetsResult(BaseModel):
+    server's local filesystem.
+
+    NOTE: files with illegal NTFS characters in their names may not be accessible to non-MacOS SMB clients.
+
+    WARNING: this value should not be changed once data is written to the SMB share. """
+
+
+class SMBWormOpt(BaseModel):
+    """ These configuration options apply to shares with the SMB_WORM_SHARE purpose. """
+    purpose: Literal[SMBSharePurpose.SMB_WORM_SHARE] = Field(exclude=True, repr=False)
+    grace_period: int = Field(default=900)
+    """ Time in seconds when write access to the file or directory is allowed. """
+    aapl_name_mangling: bool = False
+    """ If set, illegal NTFS characters commonly used by MacOS clients are stored with their native values on the SMB
+    server's local filesystem.
+
+    NOTE: files with illegal NTFS characters in their names may not be accessible to non-MacOS SMB clients.
+
+    WARNING: this value should not be changed once data is written to the SMB share. """
+
+
+class PrivateDatasetOpt(BaseModel):
+    """ These configuration options apply to shares with the PRIVATE_DATASETS_SHARE purpose. """
+    purpose: Literal[SMBSharePurpose.PRIVATE_DATASETS_SHARE] = Field(exclude=True, repr=False)
+    dataset_naming_schema: SmbNamingSchema | None = Field(default=None, examples=["%D/%u"])
+    """ The naming schema to use when `auto_dataset_creation` is specified. If you do not set a schema,
+    the server uses `%u` (username) if it is not joined to Active Directory. If the server is joined to
+    Active Directory it uses `%D/%u` (domain/username).
+
+    WARNING: ZFS dataset naming rules are more restrictive than normal path rules."""
+    auto_quota: int | None = Field(default=None, examples=[10])
+    """ Set the specified ZFS quota (in gibibytes) on new datasets. """
+    aapl_name_mangling: bool = False
+    """ If set, illegal NTFS characters commonly used by MacOS clients are stored with their native values on the SMB
+    server's local filesystem.
+
+    NOTE: files with illegal NTFS characters in their names may not be accessible to non-MacOS SMB clients.
+
+    WARNING: this value should not be changed once data is written to the SMB share. """
+
+
+class ExternalOpt(BaseModel):
+    """ These configuration options apply to shares with the EXTERNAL_SHARE purpose. """
+    purpose: Literal[SMBSharePurpose.EXTERNAL_SHARE] = Field(exclude=True, repr=False)
+    remote_path: list[NonEmptyString] = Field(examples=[
+        ['192.168.0.200\\SHARE'],
+        ['SERVER1.MYDOM.INTERNAL\\SHARE'],
+        ['SERVER1.MYDOM.INTERNAL\\SHARE, SERVER2.MYDOM.INTERNAL\\SHARE']
+    ])
+    """ This is the path to the external server and share. Each server entry must include a full domain name or IP
+    address and share name. Separate the server and share with the `\\` character.
+
+    WARNING: The SMB server and TrueNAS middleware do not check if external paths are reachable. """
```

Comment:
Probably can just state:
```suggestion
    WARNING: The TrueNAS server does not check if external paths are reachable. """
```